tx-pool/src/chunk_process.rs

```diff
@@ -0,0 +1,231 @@
+    cache::{CacheEntry, Completed},
+    ContextualWithoutScriptTransactionVerifier, ScriptVerifier, ScriptVerifyResult,
+    ScriptVerifyState, TimeRelativeTransactionVerifier, TxVerifyEnv,
+};
+use std::convert::TryInto;
+use std::{cmp, sync::Arc};
+
+use crate::component::chunk::Entry;
+use crate::component::entry::TxEntry;
+use crate::{error::Reject, service::TxPoolService};
+
+const MIN_STEP_CYCLE: Cycle = 10_000_000;
+
+type Stop = bool;
+
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub(crate) enum ProcessState {
+    Interrupt,
+    Normal,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub(crate) enum Command {
+    Suspend,
+    Continue,
+    Stop,
+}
+
+pub(crate) struct TxChunkProcess {
+    service: TxPoolService,
+    handle: Handle,
+    recv: Receiver<Command>,
+    p_state: ProcessState,
+}
+
+impl TxChunkProcess {
+    pub fn new(service: TxPoolService, handle: Handle, recv: Receiver<Command>) -> Self {
+        TxChunkProcess {
+            service,
+            handle,
+            recv,
+            p_state: ProcessState::Normal,
+        }
+    }
+
+    pub fn run(&mut self) {
+        loop {
+            select! {
+                recv(self.recv) -> res => {
+                    match res {
+                        Ok(cmd) => match cmd {
+                            Command::Continue => {
+                                self.p_state = ProcessState::Normal;
+                                let stop = self.try_process();
+                                if stop {
+                                    break;
+                                }
+                            }
+                            Command::Suspend => self.p_state = ProcessState::Interrupt,
+                            Command::Stop => {
+                                break;
+                            }
+                        },
+                        Err(_) => break,
+                    }
+                }
+                default(std::time::Duration::from_micros(1500)) => {
+                    if matches!(self.p_state, ProcessState::Normal) {
+                        let stop = self.try_process();
+                        if stop {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn try_process(&mut self) -> Stop {
+        match self.handle.block_on(self.get_front()) {
+            Some(entry) => self.process(entry),
+            None => false,
+        }
+    }
+
+    async fn get_front(&self) -> Option<Entry> {
+        self.service.chunk.write().await.pop_front()
+    }
+
+    async fn remove_front(&self) {
+        let mut guard = self.service.chunk.write().await;
+        guard.clean_front();
+    }
+
+    fn process(&mut self, entry: Entry) -> Stop {
+        self.process_inner(entry.clone()).unwrap_or_else(|e| {
+            self.handle
+                .block_on(self.service.after_process(entry.tx, entry.remote, &Err(e)));
```

Comment:
if process return error should call `remove_front` here

Reply:
https://github.com/nervosnetwork/ckb/compare/a3655fd9cec4ba037c3aa220c76ccd9b9123128d..73dc50811fc669bca01d1a63ff78d570975add3a